static/js/drafts.js

```diff
@@ -128,6 +128,19 @@ exports.restore_draft = function (draft_id) {
                               draft_copy);
     }
 
+    if (draft.type === "stream") {
+        if (draft.stream !== "") {
+            narrow.activate([{operator: "stream", operand: draft.stream},
```

Comment:
You might want to look at the comment reviews on #4615.

Pretty much, we want to make sure the stream exists before narrowing.